### PR TITLE
Move golangci-lint from circle-ci to github actions

### PR DESCRIPTION
**Pull request by sspaink**
Golangci-lint project provides a github action: https://github.com/golangci/golangci-lint-action

This pr adds a config to enable the github action (replacing circle-ci), this should be faster with caching enabled and not having to download the golangci-lint binary each time.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  golangci-pr:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
```
Comment (zak-pawel) [inline .github/workflows/golangci-lint.yml:18]: `v1.38` (or `latest`, but particular version seems to be safer) - based on https://github.com/golangci/golangci-lint/blob/master/assets/github-action-config.json

Comment (sspaink) [reply .github/workflows/golangci-lint.yml]: Good catch, I updated the version

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  golangci-pr:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.38
+          only-new-issues: true
```
Comment (zak-pawel) [inline .github/workflows/golangci-lint.yml:19]: + `args: --timeout 5m0s` ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -287,7 +278,6 @@ workflows:
                 only: /.*/
   nightly:
     jobs:
-      - 'linter'
```
Comment (zak-pawel) [inline .circleci/config.yml:290]: Don't we want to have non-failing linter run for nightly targets (for whole codebase)?

**Review by ssoroka — APPROVED** [review]
seems fine

File: .circleci/config.yml (hunk 3 above)
Comment (sspaink) [reply .circleci/config.yml:290]: good suggestion, I added a daily schedule

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (sspaink) [reply .github/workflows/golangci-lint.yml:22]: I forgot your patch with the updated config hasn't been merged yet, I added the timeout arg

**Pull request merged**